package/network/services/lldpd/files/lldpd.init

```diff
@@ -81,9 +81,25 @@ get_config_cid_ifaces() {
 
 	local _iface _ifnames=""
 	for _iface in $_ifaces; do
-		local _ifname=""
-		if network_get_device _ifname "$_iface" || [ -e "/sys/class/net/$_iface" ]; then
-			append _ifnames "${_ifname:-$_iface}" ","
+		local _l3dev=""
+		# save any "!" or "!!" prefix from the interface name
+		_suffix=${_iface##*"!"}
+		_prefix=${_iface%%"$_suffix"}
+
+		if network_get_device _l3dev "$_suffix" || [ -e "/sys/class/net/$_suffix" ]; then
+			if [ -n "$_prefix" ]; then
+				# prepend the stripped "!" or "!!" prefix here if we had one
+				append _ifnames "$_prefix${_l3dev:-$_suffix}" ","
```

Comment:
Am I missing something?  Isn't line 92 all you need instead of 90-95?  If `$_prefix` is empty then it will harmless expand to the empty string...

Reply:
Absolutely right.